streamz/core.py

```diff
@@ -647,7 +652,8 @@ def __init__(self, *children, **kwargs):
 
         children2 = [child for child in children if isinstance(child, Stream)]
 
-        Stream.__init__(self, children=children2)
+        Stream.__init__(self, children=children2, stream_name=stream_name,
+                        **kwargs)
```

Comment:
Sorry this is a mistake (I must have added the wrong thing when rebasing and resolving conflicts)
It should be:
```
Stream.__init__(self, children=children2, **kwargs)
```
and `stream_name = kwargs.pop('stream_name', None)` should be removed. It's on my branch [https://github.com/jrmlhermitte/streamz/tree/quickfix-streamname](quickfix-streamname) if you want to take a look.